src/lsstseries/ensemble.py

```diff
@@ -285,19 +285,74 @@ def prune(self, threshold=50, col_name=None):
 
         return self
 
-    def bin_sources(self, time_window=0.5, custom_aggr=None, use_map=True, **kwargs):
+    def find_day_gap_offset(self):
+        """Finds an approximation of the MJD offset for noon at the
+        observatory.
+
+        This function looks for the longest strecth of hours of the day
+        with zero observations. This gap is treated as the daylight hours
+        and the function returns the middle hour of the gap. This is used
+        for automatically finding offsets for binning.
+
+        Returns
+        -------
+        empty_hours: `list`
+            The estimated middle of the day as a floating point day. Returns
+            -1.0 if no such time is found.
+
+        Note
+        ----
+        Calls a compute on the source table.
+        """
+        if self._object_dirty:
+            self = self._sync_tables()
+
+        # Compute a histogram of observations by hour of the day.
+        hours = self._source[self._time_col].apply(
+            lambda x: np.floor(x * 24.0).astype(int) % 24, meta=pd.Series(dtype=int)
+        )
+        hour_counts = hours.value_counts().compute()
+
+        # Find the longest run of hours with no observations.
+        start_hr = 0
+        best_length = 0
+        best_mid_pt = -1
+        while start_hr < 24:
+            # Find the end of the run of zero starting at `start_hr`.
+            # Note that this might wrap around 24->0 hours.
+            end_hr = start_hr
+            while end_hr < 48 and (end_hr % 24) not in hour_counts.index:
+                end_hr += 1
+
+            # If we have found a new longest gap, record it.
+            if end_hr - start_hr > best_length:
+                best_length = end_hr - start_hr
+                best_mid_pt = (start_hr + end_hr) / 2.0
+
+            # Move to the next block.
+            start_hr = end_hr + 1
+
+        if best_length == 0:
+            return -1
+        return (best_mid_pt % 24.0) / 24.0
+
+    def bin_sources(self, time_window=1.0, offset=None, custom_aggr=None, use_map=True, **kwargs):
         """Bin sources on within a given time range to improve the estimates.
 
         Parameters
         ----------
-        time_window : `float` (optional)
+        time_window : `float`, optional
```

Comment:
I would be beneficial what is the default value when the input is not specified =

Reply:
Done.